src/Geno/C++/GUI/Modals/OpenFileModal.cpp

```diff
@@ -17,335 +17,416 @@
+						ImGui::PushStyleVar( ImGuiStyleVar_FramePadding, ImVec2( 8.0f, 5.0f ) );
+						ImGui::PushStyleColor( ImGuiCol_Border, ImVec4( 0.2f, 0.6f, 0.8f, 1.0f ) );
+						ImGui::InputTextWithHint( "##SearchFile", "Search", &m_SearchResult );
+						ImGui::PopStyleVar( 3 );
+						ImGui::PopStyleColor();
+
+						if( ImGui::IsKeyPressed( ImGui::GetKeyIndex( ImGuiKey_UpArrow ) ) )
 						{
-							m_CurrentDirectory = rDirectoryEntry;
+							m_SearchEnabled = false;
+							m_SearchResult.clear();
 						}
 					}
 				}
+				ImGui::EndChild(); //Browse With Buttons
+
+				ImGui::SameLine();
+				ImGui::SeparatorEx( ImGuiSeparatorFlags_Vertical );
+				ImGui::SameLine();
+
+				ImGui::PushStyleColor( ImGuiCol_Button, ImGui::GetStyle().Colors[ ImGuiCol_ChildBg ] );
+				ImVec2 Size = ImVec2( ImGui::GetFontSize() * m_IconSearch.GetAspectRatio() * 2.5f, ImGui::GetFontSize() * 2.1f );
+				if( ImGui::ImageButton( ( ImTextureID )m_IconSearch.GetID(), Size ) )
+				{
+					m_SearchEnabled = true;
+				}
+				ImGui::PopStyleColor();
+			}
+			ImGui::EndChild(); // Browse Panel
+			ImGui::PopStyleColor();
 
-				if( !m_DirectoryRequested )
+			ImGui::PushStyleColor( ImGuiCol_ChildBg, ImVec4( 0.15f, 0.15f, 0.15f, 0.7f ) );
+			if( ImGui::BeginChild( 6, ImVec2( 0, 0 ) ) ) // Content Browser
+			{
+				if( ImGui::BeginTable( "##FileBrowserPanel", 2, ImGuiTableFlags_Resizable | ImGuiTableFlags_ScrollY | ImGuiTableFlags_PadOuterX | ImGuiTableFlags_NoBordersInBodyUntilResize ) )
 				{
-					ImGui::Separator();
+					ImGui::TableSetupColumn( "Name", ImGuiTableColumnFlags_NoHide );
+					ImGui::TableSetupColumn( "Size", ImGuiTableColumnFlags_WidthFixed, ImGui::CalcTextSize( "Size" ).x * 2.5f );
+					ImGui::TableHeadersRow();
 
-					for( const std::filesystem::path& rFilePath : FilePaths )
+					for( const auto& rCurrentPathIt : std::filesystem::directory_iterator( m_CurrentPath ) )
 					{
-						std::string FileName = rFilePath.filename().string();
-						bool        Selected = rFilePath == m_SelectedPath;
+						std::filesystem::path Path         = rCurrentPathIt.path();
+						bool                  IsHiddenFile = Path.filename().string()[ 0 ] == '.';
 
-						if( ImGui::Selectable( FileName.c_str(), &Selected, ImGuiSelectableFlags_AllowDoubleClick ) )
+						if( !IsHiddenFile )
 						{
-							m_SelectedPath = rFilePath;
+							std::error_code ErrorCode = {};
+							if( !m_FileFilters.empty() )
+							{
+								if( m_FileFilters.find( "*" + Path.extension().string() ) == m_FileFilters.end() && !std::filesystem::is_directory( Path, ErrorCode ) )
+									continue;
+							}
 
-							// Open file immediately if file was double-clicked
-							if( ImGui::IsMouseDoubleClicked( ImGuiMouseButton_Left ) )
+							if( m_SearchEnabled && !m_SearchResult.empty() )
 							{
-								if( m_Callback )
-									m_Callback( m_SelectedPath, m_pUser );
+								if( !strstr( Path.filename().string().c_str(), m_SearchResult.c_str() ) )
```

Comment:
You can do this without C functions by using `std::string::find()`

Reply:
It's just generally better to use C++ functions over C functions, especially if we already have the structure (in this case std::string) that we need.
But this is fine for now